renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -1644,32 +1648,37 @@ bool WrappedVulkan::EndFrameCapture(void *dev, void *wnd)
     GetResourceManager()->ReleaseWrappedResource(readbackBuf);
   }
 
-  byte *jpgbuf = NULL;
-  int len = thwidth * thheight;
+  byte *buf = NULL;
+  int buflen = 0;
+  FileType thformat = FileType::JPG;
 
   if(wnd)
   {
-    jpgbuf = new byte[len];
-
-    jpge::params p;
-    p.m_quality = 80;
+    struct WriteCallbackData
+    {
+      std::vector<byte> buffer;
 
-    bool success =
-        jpge::compress_image_to_jpeg_file_in_memory(jpgbuf, len, thwidth, thheight, 3, thpixels, p);
+      static void writeData(void *context, void *data, int size)
+      {
+        WriteCallbackData *pThis = static_cast<WriteCallbackData *>(context);
+        const byte *start = static_cast<const byte *>(data);
+        pThis->buffer.insert(pThis->buffer.end(), start, start + size);
+      }
+    };
 
-    if(!success)
-    {
-      RDCERR("Failed to compress to jpg");
-      SAFE_DELETE_ARRAY(jpgbuf);
-      thwidth = 0;
-      thheight = 0;
-    }
+    WriteCallbackData callbackData;
+    stbi_write_png_to_func(&WriteCallbackData::writeData, &callbackData, thwidth, thheight, 3,
+                           thpixels, 0);
+    buf = new byte[callbackData.buffer.size()];
+    memcpy(buf, callbackData.buffer.data(), callbackData.buffer.size());
+    buflen = static_cast<int>(callbackData.buffer.size());
+    thformat = FileType::PNG;
```

Comment:
I realise your primary concern is Vulkan, but I'd prefer not to diverge the drivers like this if we can help it. If it makes things a ton easier for you to do it this way around I'd rather you either replicate this change, otherwise wait until after you've PR'd the change to move all of this code inside `CreateRDC` and maybe just do it as part of that.

Reply:
Done, apologies for the delay..